// fancy-ansi/dist/plugin.js
import { c as f } from "./colors-B3F3SizO.js";
function _(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
var c = {}, i = {};
(function(e) {
  Object.defineProperty(e, "__esModule", {
    value: !0
  }), Object.defineProperty(e, "default", {
    enumerable: !0,
    get: function() {
      return l;
    }
  });
  function u(n, t) {
    return {
      handler: n,
      config: t
    };
  }
  u.withOptions = function(n, t = () => ({})) {
    const r = function(o) {
      return {
        __options: o,
        handler: n(o),
        config: t(o)
      };
    };
    return r.__isOptionsFunction = !0, r.__pluginFunction = n, r.__configFunction = t, r;
  };
  const l = u;
})(i);
(function(e) {
  Object.defineProperty(e, "__esModule", {
    value: !0
  }), Object.defineProperty(e, "default", {
    enumerable: !0,
    get: function() {
      return n;
    }
  });
  const u = /* @__PURE__ */ l(i);
  function l(t) {
    return t && t.__esModule ? t : {
      default: t
    };
  }
  const n = u.default;
})(c);
let a = c;
var d = (a.__esModule ? a : { default: a }).default;
const s = /* @__PURE__ */ _(d), g = s(
  () => {
  },
  // do nothing
  {
    theme: {
      ansi: {
        colors: {
          ...f
        }
      }
    }
  }
);
export {
  g as default
};
